pkg/grpc/api.go

```diff
@@ -1306,12 +1307,29 @@ func (a *api) GetConfigurationAlpha1(ctx context.Context, in *runtimev1pb.GetCon
 }
 
 type configurationEventHandler struct {
+	readyCh      chan struct{}
+	readyClosed  bool
+	lock         sync.Mutex
 	api          *api
 	storeName    string
 	serverStream runtimev1pb.Dapr_SubscribeConfigurationAlpha1Server //nolint:nosnakecase
 }
 
+func (h *configurationEventHandler) ready() {
+	if !h.readyClosed {
+		close(h.readyCh)
+		h.readyClosed = true
+	}
+}
+
 func (h *configurationEventHandler) updateEventHandler(ctx context.Context, e *configuration.UpdateEvent) error {
+	// Blocks until the first message is sent
+	<-h.readyCh
```

Comment:
Important for us to exit early here if we are shutting down or something.

```suggestion
        select {
	   case <-h.readyCh:
	   case <-ctx.Done:
	     return ctx.Error()
	   }
```

Reply:
I has made this change but then realized the issue is much bigger. I'll track that in #6560